src/BraketSimulator.jl

```diff
@@ -190,91 +204,187 @@ function _compute_exact_results(d::AbstractSimulator, program::Program, qc::Int)
+    )
+        _validate_ir_results_compatibility(simulator, circuit_ir.results, Val(:JAQCD))
+        _validate_ir_instructions_compatibility(simulator, circuit_ir, Val(:JAQCD))
+        _validate_shots_and_ir_results(shots, circuit_ir.results, qubit_count)
+        operations::Vector{Instruction} = circuit_ir.instructions
+        if shots > 0 && !isnothing(circuit_ir.basis_rotation_instructions) && !isempty(circuit_ir.basis_rotation_instructions)
+            operations = vcat(operations, circuit_ir.basis_rotation_instructions)
+        end
+        inputs = get(kwargs, :inputs, Dict{String,Float64}())
+        symbol_inputs = Dict{Symbol,Number}(Symbol(k) => v for (k, v) in inputs)
+        operations = [bind_value!(Instruction(operation), symbol_inputs) for operation in operations]
+        _validate_operation_qubits(operations)
+        reinit!(simulator, qubit_count, shots)
         simulator = evolve!(simulator, operations)
+        analytic_results = shots == 0 && !isnothing(circuit_ir.results) && !isempty(circuit_ir.results)
+        results = if analytic_results
+            _compute_exact_results(simulator, circuit_ir, qubit_count)
+        else
+            Braket.ResultTypeValue[]
+        end
+        measured_qubits = get(kwargs, :measured_qubits, collect(0:qubit_count-1))
+        isempty(measured_qubits) && (measured_qubits = collect(0:qubit_count-1))
+        res = _bundle_results(results, circuit_ir, simulator, measured_qubits)
+        return res
+    end
+    simulate(simulator::AbstractSimulator, circuit_ir::Program, shots::Int; kwargs...) = simulate(simulator, circuit_ir, qubit_count(circuit_ir), shots; kwargs...)
+    function simulate(simulator::AbstractSimulator,
+                      task_specs::Vector{<:Union{Program, OpenQasmProgram}},
+                      shots::Int=0;
+                      max_parallel::Int=-1,
+                      inputs::Union{Vector{Dict{String, Float64}}, Dict{String, Float64}} = Dict{String, Float64}(),
+                      kwargs...
+                     )
+        is_single_task  = length(task_specs) == 1
+        is_single_input = inputs isa Dict || length(inputs) == 1
+        is_single_input && is_single_task && return simulate(simulator, only(task_specs), only(task_args); inputs=inputs, kwargs...)
+        if is_single_input
+            if inputs isa Dict
+                inputs = [deepcopy(inputs) for ix in 1:length(task_specs)]
+            else
+                inputs = [deepcopy(only(inputs)) for ix in 1:length(task_specs)]
+            end
+        end
+        !is_single_task && !is_single_input && (length(task_specs) != length(inputs)) && throw(ArgumentError("number of inputs ($(length(inputs))), and task specifications ($(length(task_specs))) must be equal."))
+        n_tasks = length(task_specs)
+        
+        todo_tasks_ch = Channel{Int}(ch->foreach(ix->put!(ch, ix), 1:n_tasks), n_tasks)
+
+        max_parallel_threads = max_parallel > 0 ? max_parallel : 32
```

Comment:
The reason is to avoid spawning too many short lived tasks if you have, for example, a batch of 10,000 3-qubit circuits. It's a relatively arbitrary pick that keeps Julia's scheduler from being overwhelmed if you have more than 32 cores. For machines with smaller core counts, the maximum will be `Threads.nthreads()` which should be equal to the CPU count